### DISCUSSION TITLE
upload_to_app_store updates metadata but does not update build

### DISCUSSION THREAD

**Discussion by petegivens** [fastlane/fastlane#18259] [Q&A]
I'm not sure if this is normal behavior so before I open an issue I want to see how this is supposed to work.

I'm just setting up fastlane and i've set up the following lanes:
```
  desc 'Upload the application to TestFlight'
  lane :testflight_release do
    sync_code_signing(type: 'appstore')
    build
    upload_to_testflight(username: 'xxx)
  end
  
  desc 'Upload the application to App Store'
  lane :appstore_release do
    sync_code_signing(type: 'appstore')
    increment_version_number_in_xcodeproj(bump_type: 'minor', xcodeproj: 'path/to/xcodeproj', target: 'TargetName')
    commit_version_bump(xcodeproj: '/path/to/xcodeproj', version_source: 'appstore')
    push_to_git_remote
    build
    upload_to_app_store(automatic_release: false)
  end
```

The :testflight_release lane is working as expected. However, when I call the :appstore_release lane, it's not behaving like I expected. After the upload completes, I log into appstoreconnect and see the following:
1. Under the App Store tab, the metadata seems to have been updated (The version is increased), but the selected binary under the "Build" heading has not changed.
2. Under the TestFlight tab, I see a new heading with the new version number and my build is listed there.

**tl;dr:**
When calling upload_to_app_store, TestFlight gets the new build and distributes it, but the selected build under the App Store tab does not change. Is this normal?

**Comment by AlixH** [comment]
Up
